tests/test_backends.py

```diff
@@ -237,3 +237,34 @@ def test_cumsum(self):
         result = gs.cumsum(gs.arange(10).reshape(2, 5), axis=1)
         expected = gs.array(([[0, 1, 3, 6, 10], [5, 11, 18, 26, 35]]))
         self.assertAllClose(result, expected)
+
+    def test_einsum(self):
+        np_array_1 = _np.array([[1, 1]])
+        np_array_2 = _np.array([[1, 1]])
+        array_1 = gs.array([[1, 1]])
+        array_2 = gs.array([[1, 1]])
+
+        np_result = _np.einsum('...i,...i->...', np_array_1, np_array_2)
+        gs_result = gs.einsum('...i,...i->...', array_1, array_2)
+
+        self.assertAllCloseToNp(gs_result, np_result)
+
+        np_array_1 = _np.array([[1, 1], [1, 1]])
+        np_array_2 = _np.array([[1, 1]])
+        array_1 = gs.array([[1, 1], [1, 1]])
+        array_2 = gs.array([[1, 1]])
```

Comment:
The test could be even more faithful if [1, 1] were not used everywhere. For example, if in einsum code you invert the rows or the columns of array_1 before performing einsum, there is no way to see this error in the test.

Reply:
Good point! I changed.